packages/shared/lib/time.ts

```diff
@@ -49,3 +51,40 @@ export const getBestTimeDuration = (millis: number, noDurationUnit: Duration = '
 
     return zeroTime
 }
+
+/**
+ * Converts milestone to date based on current milestone
+ *
+ * @method milestoneToDate
+ *
+ * @param {number} milestone
+ *
+ * @returns {Date}
+ */
+export const milestoneToDate = (milestone: number): Date => {
+    const currentMilestone = get(networkStatus)?.currentMilestone
+    const firstMilestoneMillis = Date.now() - currentMilestone * SECONDS_PER_MILESTONE * MILLISECONDS_PER_SECOND
+    const milestoneMillis = firstMilestoneMillis + milestone * SECONDS_PER_MILESTONE * MILLISECONDS_PER_SECOND
+    const date = new Date(milestoneMillis)
+    date.setSeconds(Math.floor(date.getSeconds() / 10) * 10)
+    return date
+}
+
+/**
+ * Gets string duration from millis in days, hours and minutes
+ *
+ * @method getDurationString
+ *
+ * @param {number} millis
+ *
+ * @returns {string}
+ */
+export const getDurationString = (millis: number): string => {
```

Comment:
Ideally we would remove the duration strings when they are `0` to avoid string like `0 days, 1 hour, 1 minute`